test/integration/user-management-api/service_contracts_controller_test.rb

```diff
@@ -25,6 +30,44 @@ def test_index
       assert xml.xpath('.//service_contracts/service-contract/id').text == @service_contract.id.to_s
     end
 
+    def test_show # to get a service contract
+      get admin_api_account_service_contract_path(account_id: @buyer.id, format: :xml, access_token: @token, id: @service_contract.id)
+      assert_response :success
+
+      xml = Nokogiri::XML::Document.parse(response.body)
+      assert xml.xpath('//service-contract/id').text == @service_contract.id.to_s
+    end
+
+    def test_success_subscribe
+      post admin_api_account_service_contracts_path(
+        account_id: @buyer.id,
+        format: :xml,
+        access_token: @token,
+        service_contract: { plan_id: @new_service_plan.id }
```

Comment:
I might be wrong, but I think we shouldn't wrap `plan_id` into `service_contract`, instead we need to verify that the controller itself wraps it properly.

Reply:
Your observation is correct. In a well-structured Rails controller, the parameters should be properly handled by the controller itself, and you should not need to explicitly wrap the plan_id into service_contract in the test.

but here its not doing so I have done it explicitily